exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2606 @@
+)
+
+func TestFuturesPositionChangePushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresPositionChangePushDataJSON)); err != nil {
+		t.Error(err)
+	}
+	if err := ku.wsHandleData([]byte(futuresPositionChangeWithChangeReasonPushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresWithdrawalAmountTransferOutAmountEventPushDataJSON = `{ "userId": "xbc453tg732eba53a88ggyt8c","topic": "/contractAccount/wallet","subject": "withdrawHold.change","data": {"withdrawHold": 5923,"currency":"USDT","timestamp": 1553842862614}}`
+
+func TestFuturesWithdrawHoldChangePushDataJSON(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresWithdrawalAmountTransferOutAmountEventPushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresAvailableBalanceChangePushData = `{ "userId": "xbc453tg732eba53a88ggyt8c","topic": "/contractAccount/wallet","subject": "availableBalance.change","data": {"availableBalance": 5923,"holdBalance": 2312,"currency":"USDT","timestamp": 1553842862614}}`
+
+func TestFuturesAvailableBalanceChangePushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresAvailableBalanceChangePushData)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresOrderMarginChangePushDataJSON = `{ "userId": "xbc453tg732eba53a88ggyt8c","topic": "/contractAccount/wallet","subject": "orderMargin.change","data": {"orderMargin": 5923,"currency":"USDT","timestamp": 1553842862614}}`
+
+func TestFuturesOrderMarginChangePushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresOrderMarginChangePushDataJSON)); err != nil {
+		t.Error(err)
+	}
+}
+
+var futuresStopOrderPushDataJSON = `{"userId": "5cd3f1a7b7ebc19ae9558591","topic": "/contractMarket/advancedOrders", "subject": "stopOrder","data": {"orderId": "5cdfc138b21023a909e5ad55","symbol": "XBTUSDM","type": "open","orderType":"stop","side":"buy","size":"1000","orderPrice":"9000","stop":"up","stopPrice":"9100","stopPriceType":"TP","triggerSuccess": true,"error": "error.createOrder.accountBalanceInsufficient","createdAt": 1558074652423,"ts":1558074652423004000}}`
+
+func TestFuturesStopOrderPushData(t *testing.T) {
+	t.Parallel()
+	if err := ku.wsHandleData([]byte(futuresStopOrderPushDataJSON)); err != nil {
```

Comment:
Please double check your time classifications here. I get `49375390-03-27 12:56:43.904 +1100 AEDT`

Just for your reference, I tested by adding the following to your conversion methods and seeing what blew up:
```go

	if k.Time().IsZero() {
		if _, file, no, ok := runtime.Caller(0); ok {
			fmt.Printf("hello  %s %v %s", file, no, k.Time())
		}
	} else if k.Time().Year() < 2017 || k.Time().Year() > 2023 {
		if _, file, no, ok := runtime.Caller(0); ok {
			return fmt.Errorf("woz  %s %v %s", file, no, k.Time())
		}
	}
```

I also found the errors returned for the following:

- TestDebtRatioChangePushData
- TestPositionStatusChangeEvent
- TestStopOrderEventPushData
- TestPublicFuturesExecutionData
- TestFuturesStopOrderPushData

Reply:
Using the code above like I described, I'm still getting incorrect times being sent:
`TestPublicFuturesExecutionData`
`TestFetchTradablePairs`
`TestFetchTicker`
`TestDebtRatioChangePushData`
`TestPositionStatusChangeEvent`
`TestStopOrderEventPushData`